drivers/sensor/ti/ina7xx/ina7xx.c

```diff
@@ -0,0 +1,391 @@
+		ina7xx_trigger_measurement(dev);
+	}
+
+	if (chan == SENSOR_CHAN_ALL) {
+		data->valid_readings_mask = 0;
+	}
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_VOLTAGE) {
+
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_VOLTAGE_BIT, false);
+		rc = ina7xx_read16(dev, INA7xx_REG_BUS_VOLTAGE, &data->v_bus);
+		if (rc < 0) {
+			LOG_ERR("Error reading bus voltage");
+			return rc;
+		}
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_VOLTAGE_BIT, true);
+	}
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_POWER) {
+
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_POWER_BIT, false);
+		rc = ina7xx_read24(dev, INA7xx_REG_POWER, &data->power);
+		if (rc < 0) {
+			LOG_ERR("Error reading power register");
+			return rc;
+		}
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_POWER_BIT, true);
+	}
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_CURRENT) {
+
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_CURRENT_BIT, false);
+		rc = ina7xx_read16(dev, INA7xx_REG_CURRENT, &data->current);
+		if (rc < 0) {
+			LOG_ERR("Error reading current register");
+			return rc;
+		}
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_CURRENT_BIT, true);
+	}
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_DIE_TEMP) {
+
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_DIE_TEMP_BIT, false);
+		rc = ina7xx_read16(dev, INA7xx_REG_DIE_TEMP, &tmp);
+		if (rc < 0) {
+			LOG_ERR("Error reading temperature register");
+			return rc;
+		}
+		WRITE_BIT(data->valid_readings_mask, INA7xx_READ_DIE_TEMP_BIT, true);
+		data->die_temp = (tmp >> INA7xx_REG_DIE_TEMP_SHIFT);
+	}
+
+	return rc;
+}
+
+static int ina7xx_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct ina7xx_data *data = dev->data;
+	int8_t sign = 1;
+
+	switch (chan) {
+	case SENSOR_CHAN_VOLTAGE:
+		if (IS_BIT_SET(data->valid_readings_mask, INA7xx_READ_VOLTAGE_BIT)) {
+			sensor_value_from_micro(val, data->v_bus * INA7xx_BUS_VOLTAGE_MUL_UV);
+		} else {
+			LOG_WRN("Last fetch did not include a voltage reading");
+			return -ENODATA;
+		}
+		break;
+	case SENSOR_CHAN_DIE_TEMP:
+		if (IS_BIT_SET(data->valid_readings_mask, INA7xx_READ_DIE_TEMP_BIT)) {
+			sensor_value_from_micro(val, data->die_temp * INA7xx_TEMP_SCALE_MG);
+		} else {
+			LOG_WRN("Last fetch did not include a die temp reading");
+			return -ENODATA;
+		}
+		break;
+	case SENSOR_CHAN_POWER:
+		if (IS_BIT_SET(data->valid_readings_mask, INA7xx_READ_POWER_BIT)) {
+			sensor_value_from_micro(val, data->power * data->power_mul_uw);
+		} else {
+			LOG_WRN("Last fetch did not include a power reading");
+			return -ENODATA;
+		}
+		break;
+	case SENSOR_CHAN_CURRENT:
+		if (IS_BIT_SET(data->valid_readings_mask, INA7xx_READ_CURRENT_BIT)) {
+			if (IS_BIT_SET((data->current), INA7xx_SIGN_BIT)) {
+				data->current = (uint16_t)(~data->current) + 1;
+				sign = -1;
+			}
+			sensor_value_from_micro(val,
+						(sign * (data->current * data->current_mul_ua)));
```

Comment:
```suggestion
			int16_t signed_current = (int16_t)data->current;
			sensor_value_from_micro(val, signed_current * (int64_t)data->current_mul_ua);
```
Replace the manual bit manipulation with a standard signed cast and use explicit type promotion to prevent undefined behavior from integer overflow.

Reply:
ok, less lines of code.